quickwit/quickwit-doc-mapper/src/query_builder.rs

```diff
@@ -33,38 +34,103 @@ pub(crate) fn build_query(
-        );
-    }
+            if needs_default_search_field(&user_input_ast)
+                && request.search_fields.is_empty()
+                && (default_field_names.is_empty() || default_field_names == [DYNAMIC_FIELD_NAME])
+            {
+                return Err(anyhow::anyhow!(
+                    "No default field declared and no field specified in query."
+                )
+                .into());
+            }
 
-    validate_requested_snippet_fields(&schema, request, &user_input_ast, default_field_names)?;
+            validate_requested_snippet_fields(
+                &schema,
+                request,
+                &user_input_ast,
+                default_field_names,
+            )?;
+
+            let search_fields = if request.search_fields.is_empty() {
+                resolve_fields(&schema, default_field_names)?
+            } else {
+                resolve_fields(&schema, &request.search_fields)?
+            };
+
+            if let Some(sort_by_field) = request.sort_by_field.as_ref() {
+                validate_sort_by_field_name(sort_by_field, &schema, Some(&search_fields))?;
+            }
 
-    let search_fields = if request.search_fields.is_empty() {
-        resolve_fields(&schema, default_field_names)?
-    } else {
-        resolve_fields(&schema, &request.search_fields)?
-    };
+            let mut query_parser =
+                QueryParser::new(schema, search_fields, QUICKWIT_TOKENIZER_MANAGER.clone());
+            query_parser.set_conjunction_by_default();
+            query_parser.parse_query(query)?
+        }
+        Some(SearchQuery::SetQuery(set_query)) => {
+            let _ = (&set_query.terms, &set_query.field_name, &set_query.tags);
+            let field_name = &set_query.field_name;
+            let field = schema
+                .get_field(field_name)
+                .ok_or_else(|| TantivyQueryParserError::FieldDoesNotExist(field_name.clone()))?;
+
+            let field_type = schema.get_field_entry(field).field_type();
+            if !field_type.is_indexed() {
+                return Err(anyhow::anyhow!("Attempted to search on not indexed field.").into());
+            }
 
-    if let Some(sort_by_field) = request.sort_by_field.as_ref() {
-        validate_sort_by_field_name(sort_by_field, &schema, Some(&search_fields))?;
-    }
+            // TODO maybe Facet could be allowed?
+            if matches!(field_type.value_type(), Type::Json | Type::Facet) {
+                return Err(
+                    anyhow::anyhow!("Attempted to search on unsuported field type.").into(),
+                );
+            }
+
+            // TODO error on any parse error instead??
+            let terms: Vec<_> = set_query
+                .terms
+                .iter()
+                // TODO not correct
+                .flat_map(|term| field_type.value_from_json(term.clone().into()).ok())
```

Comment:
is `.ok()` needed here?

```suggestion
                .flat_map(|term| field_type.value_from_json(term.clone().into()))
```

Reply:
hum, that should be a filter_map actually, but that's related to the TODO just above (which you answered), so it'll go away